smarts/core/traffic_history.py

```diff
@@ -34,13 +34,17 @@ def __init__(self, db: str):
         self._db = db
         self._db_cnxn = None
 
+    @property
+    def name(self):
+        return self._db.name[:-4]  # all history files have ".shf" extension
```

Comment:
It is better to use some filename truncation methods, rather than doing `[:-4]` to remove the file extension.